pkg/scraper/client/resource/client.go

```diff
@@ -107,17 +107,18 @@ func (kc *kubeletClient) getMetrics(ctx context.Context, url, nodeName string) (
 	if response.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("request failed, status: %q", response.Status)
 	}
-	b := kc.buffers.Get().([]byte)
+	bRef := kc.buffers.Get().(*[]byte)
+	b := *bRef
 	buf := bytes.NewBuffer(b)
```

Comment:
These 3 lines could be simplified with the following if we change the pool initialization function:

```
b := bufPool.Get().(*bytes.Buffer)
```


With that we would be able to avoid having to create a new buffer and could just reuse the one from the pool.

Reply:
Good idea, I've updated the PR.